File: app/src/main/java/com/seungjun/randomox/adapter/ReqMailListAdapter.java
package com.seungjun.randomox.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.seungjun.randomox.R;
import com.seungjun.randomox.db.LetterDBData;
import com.seungjun.randomox.db.LetterDBUtils;
import com.seungjun.randomox.view.LetterPopup;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ReqMailListAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private ArrayList<LetterDBData> letterDBData = new ArrayList<>();


    private Context context;

    private View.OnClickListener letterItemClick;

    public ReqMailListAdapter(Context context){
        this.context = context;
    }


    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {

        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.view_row_letter_list, viewGroup, false);
        return new LetterViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {

        if(letterDBData.size() > i){

            ((LetterViewHolder)viewHolder).title.setText("개발자의 " + letterDBData.get(i).get_id() +"번째 편지");
            ((LetterViewHolder)viewHolder).date.setText(letterDBData.get(i).getLetter_req_date());
            ((LetterViewHolder)viewHolder).parent.setOnClickListener(letterItemClick);

            if(letterDBData.get(i).getLetter_read().equalsIgnoreCase("Y")){
                ((LetterViewHolder)viewHolder).img.setImageDrawable(context.getResources().getDrawable(R.drawable.letter));
            }else{
                ((LetterViewHolder)viewHolder).img.setImageDrawable(context.getResources().getDrawable(R.drawable.mail));
            }

            ((LetterViewHolder)viewHolder).parent.setTag(letterDBData.get(i));

        }

    }

    public void setLetterItemClick(View.OnClickListener letterItemClick){
        this.letterItemClick = letterItemClick;
    }


    public void setLetterDBData(ArrayList<LetterDBData> list){
        letterDBData = list;
    }


    @Override
    public int getItemCount() {
        return letterDBData.size();
    }


    public class LetterViewHolder extends RecyclerView.ViewHolder{

        @BindView(R.id.view_list_title)
        TextView title;

        @BindView(R.id.view_list_date)
        TextView date;

        @BindView(R.id.view_list_row)
        LinearLayout parent;

        @BindView(R.id.view_list_img)
        ImageView img;

        public LetterViewHolder(@NonNull View itemView) {
            super(itemView);

            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/seungjun/randomox/adapter/ReqMailListAdapter.kt
package com.seungjun.randomox.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.seungjun.randomox.R
import com.seungjun.randomox.db.LetterDBData
import kotlinx.android.synthetic.main.view_row_letter_list.view.*
import java.util.*

class ReqMailListAdapter(private val context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private var letterDBData = ArrayList<LetterDBData>()

    private var letterItemClick: View.OnClickListener? = null


    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): RecyclerView.ViewHolder {

        val v = LayoutInflater.from(viewGroup.context).inflate(R.layout.view_row_letter_list, viewGroup, false)
        return LetterViewHolder(v)
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, i: Int) {

        if (letterDBData.size > i) {

            val data = letterDBData[i];

            with(viewHolder as LetterViewHolder){
                title.text = String.format("개발자의 ${data._id}번째 편지")
                date.text = data.letter_req_date
                parent.setOnClickListener(letterItemClick)

                if(data.letter_read.equals("Y", true))
                    img.setImageDrawable(context.resources.getDrawable(R.drawable.letter))
                else
                    img.setImageDrawable(context.resources.getDrawable(R.drawable.mail))

                parent.tag = data
            }
        }

    }

    fun setLetterItemClick(letterItemClick: View.OnClickListener) {
        this.letterItemClick = letterItemClick
    }


    fun setLetterDBData(list: ArrayList<LetterDBData>) {
        letterDBData = list
    }


    override fun getItemCount() = letterDBData.size


    inner class LetterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val title = itemView.view_list_title

        val date = itemView.view_list_date

        val parent = itemView.view_list_row

        val img = itemView.view_list_img

    }
}
